### DISCUSSION TITLE
Add highlighted code to context command

### DISCUSSION THREAD

**Discussion by CoderTonyB** [cline/cline#492] [Feature Requests]
A feature of the continue extension I use a lot allows me to highlight a block of code and then with a keyboard shortcut, start a chat about that code in particular.  

![image](https://github.com/user-attachments/assets/1990f35a-802a-4428-8f41-f857ca9c3321)

![image](https://github.com/user-attachments/assets/1b05c4b9-30ee-425a-92f6-98b9432009f5)
The code shows up and then I can ask questions about it.

**Comment by TomLucidor** [comment]
Which library/extension could do that for other text?

**Comment by elemental-mind** [comment]
Hey, I just created another discussion #909 , because "highlight" was not a keyword I would have used here and hence did not find it at first glance.

Could you maybe change the title to include "select" or "selection" or something for more people to find it?

**Comment by elemental-mind** [comment]
Just copying from the closed issue: 

It would be nice to have the following workflows enabled:

- I select the code in my editor
- Then I either:
  - hit a key combo to edit selection and a new chat is prefilled with my selection "@selection"
  - right click and "Cline: Edit Code" and a new chat is prefilled with my selection "@selection"
  - open a new chat and include the selection through typing "@selection"
  - open a new chat and just type "Make all the variables in my selection camel case" and Cline uses its "get_user_selection" tool to retrieve the current selection

**Comment by elemental-mind** [comment]
#631 should kind of add this feature
